datamarket/components/MyDatasets/ListDatasets.tsx

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import {
+  DatasetWithIndex,
+  FilterType,
+  SortType,
+  ViewType,
+} from "../../constants";
+import { DatasetView } from "./DatasetView";
+import SettingsIcon from "@mui/icons-material/Settings";
+import styles from "./MyDatasets.module.css";
+import ClearIcon from "@mui/icons-material/Clear";
+import AddButton from "./AddButton";
+import { sortDatasets, filterDatasets } from "@/utils/utils";
+import Sort from "./Sort";
+import View from "./View";
+import Filter from "./Filter";
+
+export const ListDatasets = ({
+  forOwnersOnly,
+  datasets,
+  setDatasets,
+  accounts,
+  liteMode,
+}: {
+  forOwnersOnly?: boolean;
+  datasets: DatasetWithIndex[];
+  setDatasets: React.Dispatch<React.SetStateAction<DatasetWithIndex[]>>;
+  accounts: string[];
+  liteMode?: boolean;
+}): React.ReactElement => {
+  const [allDatasets, setAllDatasets] =
+    React.useState<DatasetWithIndex[]>(datasets);
+  const [sidePanelHidden, setSidePanelHidden] = React.useState(true);
+  const [filter, setFilter] = React.useState<FilterType | undefined>(undefined);
+  const [sortOption, setSortOption] = React.useState<SortType | undefined>(
+    undefined
+  );
+  const [viewOption, setViewOption] = React.useState<ViewType>("Gallery");
+
+  const toggleSidePanel = (): void => {
+    setSidePanelHidden(!sidePanelHidden);
+  };
+
+  React.useEffect(() => {
+    console.log(allDatasets);
+    console.log(filter);
+    setDatasets(() => {
+      return sortDatasets(filterDatasets(allDatasets, filter), sortOption);
+    });
+  }, [sortOption, filter]);
```

Comment:
i dont like this tbh
should try to remove this and instead have this be called by an event handler
ie whenever a user clicks somethign which changes the sortOption or filter, itll call "reSortDatasets" which will perform this action.
Lmk if thats unclear

Reply:
Done